src/components/FilterSort.vue

```diff
@@ -0,0 +1,233 @@
+          aria-expanded="false"
+        >
+          Filter by date
+        </button>
+        <ul
+          class="dropdown-menu"
+          aria-labelledby="dropdownMenuButton1"
+        >
+          <div class="mx-2 my-2 dropdown-contents text-sort-filter-dropdown-text">
+            <div class="d-flex align-items-center mb-2">
+              <label
+                class="pe-2 w-25"
+                for="startDate"
+              >Start:</label>
+              <input
+                id="startDate"
+                v-model="filtersSortData.dateStart"
+                class="form-control form-control-sm text-sort-filter-dropdown-text"
+                type="date"
+                @input="emitFilterSortData"
+              >
+            </div>
+            <div class="d-flex align-items-center">
+              <label
+                for="endDate"
+                class="pe-3 w-25"
+              >End:</label>
+              <input
+                id="endDate"
+                v-model="filtersSortData.dateEnd"
+                class="form-control form-control-sm text-sort-filter-dropdown-text"
+                type="date"
+                @input="emitFilterSortData"
+              >
+            </div>
+          </div>
+        </ul>
+      </div>
+    </div>
+
+    <div class="my-2 ms-auto">
+      <button
+        type="button"
+        class="btn btn-sm btn-outline-sort-filter-clear-button-outline"
+        @click="clearAllFilters()"
+      >
+        <i class="bi bi-x-circle" />
+
+        Clear all filters
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import moment from "moment"
+import { ROLES } from "@/config/consts"
+
+export default {
+  name: "NegotiationsList",
+  props: {
+    filtersSortData: {
+      type: Object,
+      default: undefined
+    },
+    userRole: {
+      type: String,
+      required: true,
+      validator: function (value) {
+        return [ROLES.RESEARCHER, ROLES.REPRESENTATIVE].includes(value)
+      }
+    }
+  },
+  emits: ["filtersSortData"],
+  data () {
+    return {
+      filtersStatus: [
+        { value: "SUBMITTED", label: "Submitted" },
+        { value: "APPROVED", label: "Approved" },
+        { value: "DECLINED", label: "Declined" },
+        { value: "IN_PROGRESS", label: "In progress" },
+        { value: "PAUSED", label: "Paused" },
+        { value: "CONCLUDED", label: "Concluded" },
+        { value: "ABANDONED", label: "Abandoned" }
+      ],
+      sortBy: [
+        { value: "title", label: "Title" },
+        { value: "creationDate", label: "Creation Date" },
+        { value: "currentState", label: "Current State" }
+      ]
+    }
+  },
+  mounted () {
+    this.setPredifinedFiltersForStatus()
+  },
+  methods: {
+    setPredifinedFiltersForStatus () {
+      if (this.userRole === "ROLE_REPRESENTATIVE") {
+        this.filtersStatus = [{ value: "IN_PROGRESS", label: "In progress" }, { value: "ABANDONED", label: "Abandoned" }]
```

Comment:
props should not be changed directly https://michaelnthiessen.com/avoid-mutating-prop-directly

Reply:
this.filtersStatus is a data field, filtersSortData is a props

Reply:
You're right, I commented the wrong line, but also filterSortData is changed in the component and It seems it's better to avoid it